tests/mir-opt/nested_getter.outer_get.InstCombine.diff

```diff
@@ -0,0 +1,17 @@
+- // MIR for `outer_get` before InstCombine
++ // MIR for `outer_get` after InstCombine
+  
+  fn outer_get(_1: &Outer) -> u8 {
+      debug this => _1;                    // in scope 0 at $DIR/nested_getter.rs:+0:18: +0:22
+      let mut _0: u8;                      // return place in scope 0 at $DIR/nested_getter.rs:+0:35: +0:37
+      let _2: &Inner;                      // in scope 0 at $DIR/nested_getter.rs:+1:15: +1:26
+      scope 1 (inlined inner_get) {        // at $DIR/nested_getter.rs:19:5: 19:27
+          debug this => _2;                // in scope 1 at $DIR/nested_getter.rs:12:14: 12:18
+      }
+  
+      bb0: {
+          _0 = (((*_1).0: Inner).0: u8);   // scope 1 at $DIR/nested_getter.rs:13:5: 13:15
```

Comment:
Hmm, it looks like something else handled whatever this was trying to test?  Should this be `// unit-test: InstCombine` or something?

Reply:
This is supposed to be a an end-to-end test of optimizing the pattern that occurs in `Vec::as_ptr`.

But you make an interesting point here. It is odd that this doesn't currently have a diff. As far as I can tell that is because `DeadStoreElimination` suddenly takes care of this case at `-Zmir-opt-level=4`. I narrowed this down with `-Zdump-mir=all` but you can fiddle with it here: https://godbolt.org/z/K3Yzen5jf and I'll push a commit that adds the `DeadStoreElimination` diff. I think this is a bug?

Reply:
Oh wait no I mis-root-caused this. This is `ReferencePropagation` working as intended. That pass just makes a store dead, so it's not until `DeadStoreElimination` that we delete a statement.